tests/manage/test_dynamic_pvc_accessmodes_with_reclaim_policies.py

```diff
@@ -0,0 +1,353 @@
+import os
+import logging
+import pytest
+
+from ocs_ci.framework.testlib import ManageTest, tier1
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources import pvc, pod
+from ocs_ci.utility.retry import retry
+from tests import helpers
+from tests.fixtures import (
+    create_rbd_secret, create_cephfs_secret, create_ceph_block_pool,
+    create_project
+)
+
+logger = logging.getLogger(__name__)
+
+
+class BaseDynamicPvc(ManageTest):
+    """
+    Base class for Dynamic PVC creation tests
+    """
+    access_mode = None
+    storage_type = None
+    worker_nodes = None
+    expected_pvc_failure = None
+    expected_pod_failure = None
+
+    def dynamic_pvc_base(self, interface_type, reclaim_policy):
+        """
+        Base function for Dynamic PVC creation tests
+        """
+        self.interface_type = interface_type
+        self.reclaim_policy = reclaim_policy
+        self.storage_type = 'fs'
+        self.worker_nodes = self.get_all_worker_nodes()
+
+        if self.interface_type == constants.CEPHBLOCKPOOL:
+            self.interface_name = self.cbp_obj.name
+            self.secret_name = self.rbd_secret_obj.name
+
+        elif self.interface_type == constants.CEPHFILESYSTEM:
+            self.interface_name = helpers.get_cephfs_data_pool_name()
+            self.secret_name = self.cephfs_secret_obj.name
+
+        logger.info(
+            f"Creating Storage Class with reclaimPolicy: {self.reclaim_policy}"
+        )
+        self.sc_obj = helpers.create_storage_class(
+            interface_type=self.interface_type,
+            interface_name=self.interface_name,
+            secret_name=self.secret_name,
+            reclaim_policy=self.reclaim_policy
+        )
+
+        logger.info(f"Creating PVC with accessModes: {self.access_mode}")
+        self.pvc_obj = helpers.create_pvc(
+            access_mode=self.access_mode, sc_name=self.sc_obj.name,
+            namespace=self.namespace, size='10Gi', wait=True
+        )
+
+    def run_io(self, pod_obj, file_name):
```

Comment:
seems like a handy function, please move to pod.py with a small tweak:
The function should wait and print the results, not starting fio

Reply:
Done.
Added get_fio_rw_iops() in pod.py